workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/GeneratorDetails.cs

```diff
@@ -0,0 +1,18 @@
+using Improbable.Gdk.CodeGeneration.Utils;
+
+namespace Improbable.Gdk.CodeGeneration.Model.Details
+{
+    public abstract class GeneratorDetails
+    {
+        public readonly string Name;
+        public readonly string Namespace;
+        public readonly string NamespacePath;
+
+        protected GeneratorDetails(string rawName, string package)
+        {
+            Name = rawName;
+            Namespace = Formatting.CapitaliseQualifiedNameParts(package);
+            NamespacePath = Formatting.GetNamespacePath(package);
+        }
```

Comment:
Suggestion - add methods (or properties) like:

```
public string CamelCaseName() 
{
}

public string FullQualifiedName() 
{
}
```

to this base class since you can implement them here. Then all children get these for free